example/tests/example_test.py

```diff
@@ -1,22 +1,60 @@
 import unittest
 
-from snapstack import Runner
+from snapstack import Plan, Step, Setup, Cleanup
 
 
 class ExampleTest(unittest.TestCase):
 
-    def test(self):
-        r = Runner(
+    def test_example(self):
+        '''
+        _test_example_
+
+        Basic test, which installs and tests and example
+        snap, after letting the Plan object build the default
+        snapstack.
+
+        Note: this will install snapstack on your system!
+
+        '''
+        example = Step(
             snap='example',  # Name of the snap
-            location='./tests/',  # Parent location for tests and files
-                                  # below. {local} is a magic string that
-                                  # points at the 'tests' directory.
+            script_loc='./tests/',  # Parent location for tests and files
+                                    # below. {local} is a magic string that
+                                    # points at the 'tests' directory.
             tests=['example.sh'],  # Test the snap
             files=['example.json'],  # Config files
-            base=None  # Don't override default Base.
+            snap_store=False  # Build from local source, rather than
+                              # install from the snap store.
         )
+        plan = Plan(tests=[example])
+        plan()
+
+    def test_custom_base(self):
+        '''
+        _test_custom_base_
+
+        Reference code for adding and removing stuff from our Setup
+        and Cleanup.
+
+        '''
+        setup = Setup()
+        cleanup = Cleanup()
+
+        # Add a step.
+        example = Step(
+            snap='example',
+            script_loc='./tests/',
+            tests=['example.sh'],
+            files=['example.json'],
+        )
+        setup.add_steps(('example', example))
+
+        # Remove steps.
+        setup.remove_steps(
+            'keystone', 'nova', 'neutron', 'glance', 'nova_hypervisor',
+            'neutron_ext_net')
 
-        try:
-            r.run()
-        finally:
-            r.cleanup()
+        # Run with our changes (this will download the config files,
+        # but then install nothing but the example snap).
+        plan = Plan(tests=[], setup=setup.steps(), cleanup=cleanup.steps())
```

Comment:
I like what you're doing with Setup() and Cleanup(). I think there were 2 issues that I noticed.

One thing I noticed is that while the default setup can is cleaned up by the default cleanup. There's no cleanup argument for the tests argument.  Ie. it seems like the current snap's tests need an argument for cleanup.

The other thing is that I found the usage between the first and 2nd test to be a little awkward, in that example was added to the setup argument, whereas in the first test it was passed in under the test argument, but that is probably just a usage difference.